addons/udes_stock/tests/test_push_from_drop.py

```diff
@@ -0,0 +1,253 @@
+        """
+        Push = self.env['stock.rule']
+        input_zone = self.env.ref('stock.stock_location_company')
+
+        # create new push from the Input zone.
+        Push.create({
+            "name": "TestPushMultiRoute",
+            "route_id": self.route_in.id,
+            "sequence": 40,
+            "location_src_id": input_zone.id,
+            "location_id": self.location_qc_zone.id,
+            "picking_type_id": self.picking_type_putaway.id,
+            "u_push_on_drop": True,
+            "action": "push",
+        })
+        # Damaged route found correctly
+        res = Push.get_path_from_location(self.received_damaged_location)
+        self.assertEqual(res, self.push_damaged_putaway)
+
+        # Non-damaged route found correctly
+        res = Push.get_path_from_location(self.received_location)
+        self.assertEqual(res, self.push_putaway)
+
+
+class TestCreateMovesForPush(PushFromDropBase):
+    """Tests for the stock.move._create_moves_for_push method in
+    isolation"""
+    def setUp(self):
+        """Create a picking to work with for tests"""
+        super(TestCreateMovesForPush, self).setUp()
+        products_info = [{'product': self.apple, 'uom_qty': 10},
+                         {'product': self.banana, 'uom_qty': 99}]
+        self.goods_in = self.create_picking(self.picking_type_goods_in,
+                                            products_info=products_info)
+        self.goods_in.action_confirm()
+        self.goods_in.action_assign()
+        self.move_lines = self.goods_in.move_line_ids
+        for ml in self.move_lines:
+            ml.write({'qty_done': ml.product_uom_qty})
+
+    def test01_correct_move_qty(self):
+        """Test that created moves have the correct quantity"""
+        Move = self.env['stock.move']
+        res = Move._create_moves_for_push(self.push_damaged_putaway,
+                                          self.move_lines)
+        self.assertEqual(len(res), 2)
+        new_apple_move = res.filtered(lambda m: m.product_id.id == self.apple.id)
+        new_banana_move = res.filtered(lambda m: m.product_id.id == self.banana.id)
+
+        self.assertEqual(new_apple_move.product_uom_qty, 10)
+        self.assertEqual(new_banana_move.product_uom_qty, 99)
+
+    def test02_correct_move_locations(self):
+        """Test that created moves have the correct src and dest locations"""
+        Move = self.env['stock.move']
+        res = Move._create_moves_for_push(self.push_damaged_putaway,
+                                          self.move_lines)
+
+        self.assertEqual(len(res), 2)
+        for move in res:
+            self.assertEqual(move.location_id, self.received_damaged_location)
+            self.assertEqual(move.location_dest_id, self.location_qc_zone)
+
+    def test03_correct_move_orig_info(self):
+        """Test that created moves set move_orig_ids correctly"""
+        Move = self.env['stock.move']
+
+        moves = self.goods_in.mapped('move_lines')
```

Comment:
do we need the mapped here?

Reply:
@jrbirch99 in Odoo 14 you don't need to do mapped for relation fields (i.e. `one2many`, `many2one` and `many2many`).

`mapped` is still needed for other field types though such as `char`.